src/app/offers/offer-details/offer-details.component.ts

```diff
@@ -76,21 +75,18 @@ export class OfferDetailsComponent implements OnInit {
             switchMap(shouldLoadUser => shouldLoadUser ? this.userService.user$ : EMPTY),
             take(1)
         );
-        this.portfolio$ = this.portfolioService.getPortfolio().pipe(
-            map(res => res.portfolio));
+
+        this.investmentTotal$ = this.portfolioService.getPortfolio().pipe(
+            switchMap(res => res.portfolio),
+            find(item => item.project.uuid === this.route.snapshot.params.id),
+            map(portfolio => portfolio?.investment)
+        );
     }
 
     ngOnInit() {
         if (this.route.snapshot.params.isOverview) {
             this.isOverview = true;
         }
-        this.portfolio$.subscribe(res => {
-            res.forEach(elem => {
-                if (elem.project.uuid === this.route.snapshot.params.id) {
-                    this.transactionTotal = elem.investment;
-                }
-            });
-        });
```

Comment:
This will subscribe to portfolio also in overview and in invest screen, which we don't want. Consider using plain observable and subscribe to it inside template (async pipe).